project/driver_controls/inc/input_event.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+// Define common typedefs for the various FSMs to use.
+
+#include "event_queue.h"
+
+// Typedef for the FSM arbitration functions
+typedef bool (*InputEventCheck)(Event *e);
```

Comment:
I needed to keep it visible to the FSMs since InputEventCheck is what the context pointer is cast to in the FSM output functions.